integration-cli/docker_cli_build_test.go

```diff
@@ -239,9 +239,18 @@ func TestBuildEnvironmentReplacementEnv(t *testing.T) {
 
 	_, err := buildImage(name,
 		`
-  FROM scratch
+  FROM busybox
   ENV foo foo
   ENV bar ${foo}
+  ENV abc1='$foo'
+  ENV env1=$foo env2=${foo} env3="$foo" env4="${foo}"
+  RUN [ "$abc1" = '$foo' ] && (echo "$abc1" | grep foo)
```

Comment:
I needed to make sure that $foo wasn't replaced so grep'ing for 'foo' seemed like a good test.